OutOfSchool/OutOfSchool.WebApi.Tests/Services/Database/SensitiveWorkshopsServiceDBTests.cs

```diff
@@ -0,0 +1,211 @@
+
+        workshopRepository = new WorkshopRepository(dbContext);
+        dateTimeRangeRepository = new Mock<IEntityRepositorySoftDeleted<long, DateTimeRange>>();
+        roomRepository = new Mock<IEntityRepositorySoftDeleted<Guid, ChatRoomWorkshop>>();
+        teacherService = new Mock<ITeacherService>();
+        logger = new Mock<ILogger<WorkshopService>>();
+        workshopImagesMediator = new Mock<IImageDependentEntityImagesInteractionService<Workshop>>();
+        mapper = TestHelper.CreateMapperInstanceOfProfileTypes<CommonProfile, MappingProfile>();
+        providerAdminRepository = new Mock<IProviderAdminRepository>();
+        averageRatingServiceMock = new Mock<IAverageRatingService>();
+        providerRepositoryMock = new Mock<IProviderRepository>();
+        currentUserServiceMock = new Mock<ICurrentUserService>();
+        ministryAdminServiceMock = new Mock<IMinistryAdminService>();
+        regionAdminServiceMock = new Mock<IRegionAdminService>();
+        codeficatorServiceMock = new Mock<ICodeficatorService>();
+
+        sensitiveWorkshopService =
+            new WorkshopService(
+                workshopRepository,
+                dateTimeRangeRepository.Object,
+                roomRepository.Object,
+                teacherService.Object,
+                logger.Object,
+                mapper,
+                workshopImagesMediator.Object,
+                providerAdminRepository.Object,
+                averageRatingServiceMock.Object,
+                providerRepositoryMock.Object,
+                currentUserServiceMock.Object,
+                ministryAdminServiceMock.Object,
+                regionAdminServiceMock.Object,
+                codeficatorServiceMock.Object);
+
+        institutionId = Guid.Parse("f72f6404-0d4a-4a6a-bee1-f1a660f0f556");
+        Seed();
+    }
+
+    [TearDown]
+    public void Dispose()
+    {
+        dbContext.Dispose();
+    }
+
+    [Test]
+    public async Task FetchByFilterForAdmins_FilterIsNull_ShouldBuildPredicateAndReturnAllEntities()
+    {
+        // Arrange
+        var workshopDto = await MapWorkshopsToDtos();
+        var expectedResult = new SearchResult<WorkshopDto>()
+        {
+            TotalAmount = workshopDto.Count,
+            Entities = workshopDto,
+        };
+
+        // Act
+        var result = await sensitiveWorkshopService.FetchByFilterForAdmins().ConfigureAwait(false);
+
+        // Assert
+        result.Should().BeEquivalentTo(expectedResult);
+    }
+
+    [Test]
+    public async Task FetchByFilterForAdmins_ByFilterCriteria_ShouldBuildPredicateAndReturnMatchEntities()
+    {
+        // Arrange
+        var workshopDto = await MapWorkshopsToDtos();
+        var filter = new WorkshopFilterAdministration()
+        {
+            InstitutionId = institutionId,
+            SearchString = "keyWord",
+            Size = 100,
+        };
+        var expectedEntities = new List<WorkshopDto> {workshopDto[0] };
```

Comment:
(Test logic) What am I missing? Why test expects 0 entities? As I can see at least one entity with `keyWord` is present in seeded data.

Reply:
The test expects the first element from the list of DTOs.

Reply:
Indeed. I need more sleep. Thank you.